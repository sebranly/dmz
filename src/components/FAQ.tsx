import classnames from 'classnames';
import * as React from 'react';
import { CURRENT_SEASON } from '../constants/game';
import { YOUTUBE_REFERENCE } from '../constants/general';
import { questionsAnswers } from '../data/faq';
import { QuestionAnswer } from '../types';
import { displayWithTwoDigits } from '../utils/display';

const FAQ = () => {
  const [faq, setFaq] = React.useState(questionsAnswers);

  const onToggle = (questionString: string) => {
    const newFaq = faq.map((qa: QuestionAnswer) => {
      const { answer, isNew, question, shown, yt } = qa;
      const isClicked = question === questionString;
      const shownValue = isClicked ? !shown : false;
      return { question, answer, isNew: !!isNew, shown: shownValue, yt: !!yt };
    });

    setFaq(newFaq);
  };

  return (
    <div id="faq">
      <h3>FAQ</h3>
      {faq.map((qa: QuestionAnswer) => {
        const { answer, isNew, question, shown, yt } = qa;

        const questionSuffix = shown ? '-' : '+';
        const questionBis = `${question} [${questionSuffix}]`;
        const classnamesQuestionAndAnswer = classnames(
          { 'margin-top-10': !shown },
          { 'margin-bottom-20 margin-top-20': shown }
        );

        return (
          <div key={question} className={classnamesQuestionAndAnswer}>
            <div onClick={() => onToggle(question)} className="question">
              {isNew && <div className="inline color-orange">{'[New] '}</div>}
              {questionBis}
            </div>
            {shown && (
              <div className="answer">
                {answer.map((answerParagraph: string, index: number) => {
                  const classnamesParagraph = classnames({ inline: index === answer.length - 1 && yt });
                  return (
                    <p className={classnamesParagraph} key={index}>
                      {answerParagraph}
                    </p>
                  );
                })}
                {yt && (
                  <>
                    {' '}
                    <a
                      className="yt-video"
                      href={YOUTUBE_REFERENCE}
                      rel="noopener noreferrer"
                      title="YouTube reference for conversion functions"
                      target="_blank"
                    >
                      {`watch S${displayWithTwoDigits(CURRENT_SEASON)} video`}
                    </a>
                  </>
                )}
              </div>
            )}
          </div>
        );
      })}
    </div>
  );
};

export { FAQ };
