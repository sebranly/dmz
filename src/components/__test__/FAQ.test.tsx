import * as React from 'react';
import { render } from '@testing-library/react';

import { FAQ } from '../FAQ';

test('FAQ', () => {
  const createComponent = () => <FAQ />;

  const { container } = render(createComponent());
  expect(container.childNodes).toMatchSnapshot();
});
