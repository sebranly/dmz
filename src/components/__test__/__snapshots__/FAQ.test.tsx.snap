// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAQ 1`] = `
NodeList [
  <div
    id="faq"
  >
    <h3
      class="font-bold my-5 text-xl text-lime-500"
    >
      FAQ
    </h3>
    <div
      class="text-base lg:text-lg mt-2.5"
    >
      <div
        class="text-left sm:text-center cursor-pointer font-bold mb-2 text-lime-400 hover:text-lime-500"
      >
        What is DMZ? [+]
      </div>
    </div>
    <div
      class="text-base lg:text-lg my-5"
    >
      <div
        class="text-left sm:text-center cursor-pointer font-bold mb-2 text-lime-400 hover:text-lime-500"
      >
        What is this website? [-]
      </div>
      <div
        class="text-left sm:text-justify mb-5"
      >
        <p
          class="mb-5"
        >
          DMZ: Currency to Cooldown Converter is an unofficial website which focuses on getting your lost, insured weapons back as efficiently as possible. In the game, dying on the battlefield with an insured weapon means that the weapon gets locked behind a cooldown timer. By default, the player has to wait for a certain amount of hours before selecting this same weapon.
        </p>
        <p
          class="mb-5"
        >
          The player can make this process faster by using money to reduce said cooldown timer. This website provides money-to-time and time-to-money conversions for these lost, insured weapons.
        </p>
      </div>
    </div>
    <div
      class="text-base lg:text-lg mt-2.5"
    >
      <div
        class="text-left sm:text-center cursor-pointer font-bold mb-2 text-lime-400 hover:text-lime-500"
      >
        What are the hourly rates? [+]
      </div>
    </div>
    <div
      class="text-base lg:text-lg mt-2.5"
    >
      <div
        class="text-left sm:text-center cursor-pointer font-bold mb-2 text-lime-400 hover:text-lime-500"
      >
        What are dead drops? [+]
      </div>
    </div>
    <div
      class="text-base lg:text-lg mt-2.5"
    >
      <div
        class="text-left sm:text-center cursor-pointer font-bold mb-2 text-lime-400 hover:text-lime-500"
      >
        <div
          class="inline text-amber-500 hover:text-amber-600"
        >
          [New] 
        </div>
        Where are dead drops? [+]
      </div>
    </div>
    <div
      class="text-base lg:text-lg mt-2.5"
    >
      <div
        class="text-left sm:text-center cursor-pointer font-bold mb-2 text-lime-400 hover:text-lime-500"
      >
        What about items? [+]
      </div>
    </div>
    <div
      class="text-base lg:text-lg mt-2.5"
    >
      <div
        class="text-left sm:text-center cursor-pointer font-bold mb-2 text-lime-400 hover:text-lime-500"
      >
        I don't understand the use-case(s) [+]
      </div>
    </div>
    <div
      class="text-base lg:text-lg mt-2.5"
    >
      <div
        class="text-left sm:text-center cursor-pointer font-bold mb-2 text-lime-400 hover:text-lime-500"
      >
        <div
          class="inline text-amber-500 hover:text-amber-600"
        >
          [New] 
        </div>
        I think your formulae are incorrect [+]
      </div>
    </div>
    <div
      class="text-base lg:text-lg mt-2.5"
    >
      <div
        class="text-left sm:text-center cursor-pointer font-bold mb-2 text-lime-400 hover:text-lime-500"
      >
        <div
          class="inline text-amber-500 hover:text-amber-600"
        >
          [New] 
        </div>
        What changed since Season 01? [+]
      </div>
    </div>
    <div
      class="text-base lg:text-lg mt-2.5"
    >
      <div
        class="text-left sm:text-center cursor-pointer font-bold mb-2 text-lime-400 hover:text-lime-500"
      >
        <div
          class="inline text-amber-500 hover:text-amber-600"
        >
          [New] 
        </div>
        Is there a time difference between the maps? [+]
      </div>
    </div>
    <div
      class="text-base lg:text-lg mt-2.5"
    >
      <div
        class="text-left sm:text-center cursor-pointer font-bold mb-2 text-lime-400 hover:text-lime-500"
      >
        <div
          class="inline text-amber-500 hover:text-amber-600"
        >
          [New] 
        </div>
        Why is the maximum number of players high on this website? [+]
      </div>
    </div>
    <div
      class="text-base lg:text-lg mt-2.5"
    >
      <div
        class="text-left sm:text-center cursor-pointer font-bold mb-2 text-lime-400 hover:text-lime-500"
      >
        Can I bake cakes with this website? [+]
      </div>
    </div>
    <div
      class="text-base lg:text-lg mt-2.5"
    >
      <div
        class="text-left sm:text-center cursor-pointer font-bold mb-2 text-lime-400 hover:text-lime-500"
      >
        What about cookies? [+]
      </div>
    </div>
  </div>,
]
`;
